pulsar-function-go/pf/instance.go

```diff
@@ -374,17 +409,201 @@ func (gi *goInstance) healthCheck() *pb.HealthCheckResult {
+		// Should we panic here instead of report an error since it reflects a code problem, not a user problem?
+	}
+	metricFunc := gi.getMatchingMetricFunc()
+	matchingMetric := getFirstMatch(fiteredMetricFamilies[0].Metric, metricFunc)
+	return *matchingMetric
+}
+
+func (gi *goInstance) getTotalReceived() float32 {
+	// "pulsar_function_" + "received_total", NewGaugeVec.
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + TotalReceived)
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+func (gi *goInstance) getTotalProcessedSuccessfully() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + TotalSuccessfullyProcessed)
+	// "pulsar_function_" + "processed_successfully_total", NewGaugeVec.
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+func (gi *goInstance) getTotalSysExceptions() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + TotalSystemExceptions)
+	// "pulsar_function_"+ "system_exceptions_total", NewGaugeVec.
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+
+func (gi *goInstance) getTotalUserExceptions() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + TotalUserExceptions)
+	// "pulsar_function_" + "user_exceptions_total", NewGaugeVec
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+
+func (gi *goInstance) getAvgProcessLatency() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + ProcessLatencyMs)
+	// "pulsar_function_" + "process_latency_ms", SummaryVec.
+	count := metric.GetSummary().SampleCount
+	sum := metric.GetSummary().SampleSum
+	if *count <= 0.0 {
+		return 0.0
+	}
+	return float32(*sum) / float32(*count)
+}
+
+func (gi *goInstance) getLastInvocation() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + LastInvocation)
+	// "pulsar_function_" + "last_invocation", GaugeVec.
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+
+func (gi *goInstance) getTotalProcessedSuccessfully1min() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + TotalSuccessfullyProcessed1min)
+	// "pulsar_function_" + "processed_successfully_total_1min", GaugeVec.
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+
+func (gi *goInstance) getTotalSysExceptions1min() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + TotalSystemExceptions1min)
+	// "pulsar_function_" + "system_exceptions_total_1min", GaugeVec
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+
+func (gi *goInstance) getTotalUserExceptions1min() float32 {
+	metric := gi.getMatchingMetricFromRegistry(PulsarFunctionMetricsPrefix + TotalUserExceptions1min)
+	// "pulsar_function_" + "user_exceptions_total_1min", GaugeVec
+	val := metric.GetGauge().Value
+	return float32(*val)
+}
+
+/*
+func (gi *goInstance) get_avg_process_latency_1min() float32 {
```

Comment:
Do we still need this?

Reply:
@jiazhai This is a method that isn't available due to the Prometheus architecture change they rolled out to their Go library. So, we need to find another way to capture the 1 min metrics. The comment was to note that this feature is currently missing.